mailbox/event/event-rabbitmq/src/test/java/org/apache/james/mailbox/events/RabbitMQEventBusTest.java

```diff
@@ -791,6 +791,18 @@ void dispatchShouldPersistEventsWhenDispatchingGroupsGetErrorMultipleTimes() {
             assertThat(dispatchingFailureEvents()).containsExactly(EVENT, EVENT_2);
         }
 
+        @Test
+        void dispatchShouldPersistEventsWhenDispatchingTheSameEventGetErrorMultipleTimes() {
+            EventCollector eventCollector = eventCollector();
+            eventBus().register(eventCollector, GROUP_A);
+
+            rabbitMQExtension.getRabbitMQ().pause();
+            doQuietly(() -> eventBus().dispatch(EVENT, NO_KEYS).block());
+            doQuietly(() -> eventBus().dispatch(EVENT, NO_KEYS).block());
+
+            assertThat(dispatchingFailureEvents()).containsExactly(EVENT, EVENT);
```

Comment:
What happens then when it's healthy again and we replay them successfully? we replay twice the same event? is there any consequence to that?

Reply:
we send the same event twice, obviously, the event should be played twice.

> What happens then when it's healthy

If it's healthy, that means no event in deadletter, so you can not replay because nothing in there. I don't get your question

Reply:
> is there any consequence to that?

Our failure mode is "at least once". Double delivery is thus acceptable.